pkg/cache/sql/informer/factory/informer_factory.go

```diff
@@ -149,3 +164,23 @@ func (f *CacheFactory) Reset() error {
 
 	return nil
 }
+
+// checkWatchability checks whether the specified gvk can be watched by this cache factory
+func (f *CacheFactory) checkWatchability(gvk schema.GroupVersionKind) error {
+	apiResourceList, err := f.discovery.ServerResourcesForGroupVersion(gvk.GroupVersion().String())
+	if err != nil {
+		return fmt.Errorf("failed to check watchability of %v, ServerResourcesForGroupVersion returned: %w", gvk, err)
+	}
+
+	for _, resource := range apiResourceList.APIResources {
+		if resource.Kind == gvk.Kind {
+			for _, verb := range resource.Verbs {
+				if verb == "watch" {
+					return nil
+				}
+			}
+			return &UnwatchableError{gvk}
```

Comment:
You could use `slices.Contains(resource.Verbs, "watch")` ?

And I don't think you need to return `&WatchableError{gvk}` here, because if none of the `apiResourceList.APIResources` have a "watch" verb, then it will just drop out to the return at the end of the function?

Reply:
Agree with the points above. I think that we should still break at this point to avoid iterating over the other GVKs once we have located the target one.